crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,441 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use self::flex::FlexLayout;
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Layout {
+    /// Use the absolute position and size specified
+    PositionLayout,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm
+    ///
+    /// As implemented by [`taffy`]: some bugs or limitations may exist;
+    /// [please file an issue!](https://github.com/DioxusLabs/taffy/issues)
+    FlexLayout(FlexLayout),
+}
```

Comment:
`LayoutStrategy` has now become `Layout`. 

And since the `FlexLayout` struct is a payload in the enum, we no longer need the additional `flex_layout` field on the Node's.